e2e/support/config.js

```diff
@@ -61,6 +63,22 @@ const defaultConfig = {
       });
     }
 
+    on("before:run", () => {
+      console.log("🧹 Clean up the existing Cypress artifacts 🧹");
+      fs.rmSync(path.join(config.projectRoot + "/cypress"), {
+        recursive: true,
+        force: true,
+      });
+    });
+
+    on("after:run", results => {
+      if (results.totalFailed > 0) {
+        const reportDir = path.join(config.projectRoot + "/cypress/reports");
+
+        console.log("📄 Generate unified Mocha report 📄");
+        generateReport(reportDir);
+      }
+    });
```

Comment:
The official example suggests to do this as part of the custom runner wrapper instead. However, `on:after:run` hook has all the info it needs to do this job correctly.
https://github.com/Antontelesh/mochawesome-merge?tab=readme-ov-file#cypress

The only downside (as stated in the PR description) is that failure in this hook will not terminate the main Cypress process. It will need to be killed either by a timeout (CI) or manually (local run).

Reply:
I’m wondering if it wouldn’t rather make sense to merge mochawesome reports as a separate CI step instead.

Reply:
That could work as well.
But do you think people will need this when running Cypress locally? Not sure how many people rely on these reports (I bet it's close to zero).

Reply:
I doubt anyone needs reports locally. How many tests are you going to run on a local instance anyway? not enough to need a report in my opinion. Ideally I think the whole CI tooling should not even load when running locally

Reply:
Awesome, that's even better. I'll do it as a CI step specifically.

Do you consider on before run's removal of artifacts to be a part of "the whole CI tooling"?